src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,822 @@ def test_rotate_img_metadata(self):
+        }
+
+    def test_get_bounding_box_normal(self):
+        """
+        Verify that using a 5D shaped DA will return the right bounding box.
+        """
+        # create a DA with random values
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        output_pos = img.getBoundingBox(new_da)
+
+        # set the positions to check for
+        pos_x = new_da.metadata[model.MD_POS][0]
+        pos_y = new_da.metadata[model.MD_POS][1]
+
+        # check if the min position of the bounding box is the same as half the 
+        # size of the DA calculating from the middle point on the 0-axis downwards
+        expected_pos_x_min = pos_x - self.size[-1] * self.md[model.MD_PIXEL_SIZE][0] / 2
+        self.assertEqual(output_pos[0], expected_pos_x_min)
+
+        # check if the max position of the bounding box is the same as half the
+        # size of the DA calculating from the middle point on the 0-axis upwards
+        expected_pos_x_max = pos_x + self.size[-1] * self.md[model.MD_PIXEL_SIZE][0] / 2
+        self.assertEqual(output_pos[2], expected_pos_x_max)
+
+        # check if the min position of the bounding box is the same as half the 
+        # size of the DA calculating from the middle point on the 1-axis downwards
+        expected_pos_y_min = pos_y - self.size[-2] * self.md[model.MD_PIXEL_SIZE][1] / 2
+        self.assertEqual(output_pos[1], expected_pos_y_min)
+
+        # check if the max position of the bounding box is the same as half the
+        # size of the DA calculating from the middle point on the 1-axis upwards
+        expected_pos_y_max = pos_y + self.size[-2] * self.md[model.MD_PIXEL_SIZE][1] / 2
+        self.assertEqual(output_pos[3], expected_pos_y_max)
+
+    def test_get_bounding_box_incomplete_metadata(self):
+        """
+        Verify that removing PIXEL_SIZE in the metadate DA will fail and a LookupError exception is raised.
+        """
+        # create a DA with random values
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        # remove the PIXEL_SIZE key
+        del self.md[model.MD_PIXEL_SIZE]
+
+        # force a LookupError
+        with self.assertRaises(LookupError):
+            img.getBoundingBox(new_da)
+
+    def test_get_bounding_box_wrong_arrayshape(self):
+        """
+        Verify that using a wrong shaped DA will fail and if an IndexError exception is raised.
+        """
+        # create a wrong shaped DA with random values
+        self.size = (4000, 3500)
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        # force an IndexError due to wrong DA shape
+        with self.assertRaises(IndexError):
+            img.getBoundingBox(new_da)
+
+
+class TestAssembleZCube(unittest.TestCase):
+
+    def setUp(self):
+        # create a 2D size and metadata with necessary items
+        self.size = (3200, 3200)
```

Comment:
Maybe pick something a little smaller.  For the test cases, I don't think the size matters much, but as you have quite a few Z, that's going to consume quite a bit of memory.
You should also pick something not square.
=> `size = 400, 300`